src/ucp/core/ucp_am.c

```diff
@@ -359,6 +359,17 @@ ucp_am_fill_first_footer(ucp_am_first_ftr_t *ftr, ucp_request_t *req)
     ftr->total_size = req->send.length;
 }
 
+static UCS_F_ALWAYS_INLINE void ucp_am_put_user_header(ucp_request_t *req)
```

Comment:
name looks confusing, maybe `ucp_am_put_cached_header`?

Reply:
Header buffer is from mpool. I've chose this name to hint that (to resemble to ucs_mpool_put).
I don't know if cached is better... it's confusing... maybe just ucp_am_release_user_header ?

Reply:
need to align with API name (I noticed @shamisp 's comment), so if it's copy, then `ucp_am_put_copy_header`

Reply:
Maybe ucp_am_put_header_copy? you are right that the name of the flag in API is ...COPY_HEADER, but "put_copy_header" is an odd term.